pkg/configprovider/providerutil/utils.go

```diff
@@ -0,0 +1,44 @@
+package providerutil
+
+import (
+	"go.opentelemetry.io/collector/confmap"
+)
+
+// Removes the key(path passed in keys array) from sourceMap if same key path is present in MergeMap
+func removeMatchingKeyFromSrcMap(srcMap map[string]interface{}, mergeMap map[string]interface{}, keys []string) map[string]interface{} {
+
+	if len(keys) == 0 {
+		return srcMap
+	}
+
+	currentKey := keys[0]
+	if len(keys) == 1 { // Got Leaf key, if leaf key exists in both maps, remove it from source map
+		_, existInSrc := srcMap[currentKey]
+		_, existInMerge := mergeMap[currentKey]
+		// If leaf key exists in both maps, remove from source map
+		if existInSrc && existInMerge {
+			delete(srcMap, currentKey)
+		}
+		return srcMap
+	}
+	// More levels to go, descend into child if current key present in both maps
+	srcNestedMap, isCurrKeyInSrcMap := srcMap[currentKey].(map[string]interface{})
+	mergeNestedMap, isCurrKeyInMergeMap := mergeMap[currentKey].(map[string]interface{})
+	if isCurrKeyInSrcMap && isCurrKeyInMergeMap {
+		removeMatchingKeyFromSrcMap(srcNestedMap, mergeNestedMap, keys[1:])
```

Comment:
is this effectively removing the first entry from a map? How are we ensuring that the keys are in the right order, i.e sorted by their file name?